install.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+set -e
+
+version="2.2.0"
+
+# Do nothing if echidna is already installed
+if echidna-test --version > /dev/null 2>&1
+then
+  echo "WARNING: $(echidna-test --version) is installed but is out-of-date, we recommend removing it"
+fi
+
+# Do nothing if echidna is already installed
+if echidna --version > /dev/null 2>&1
+then
+  echo "$(echidna --version) is already installed" && exit
+  # TODO: check that the version is up to date, otherwise offer to upgrade
+fi
+
+# In the unlikely scenario that python3 isn't available, let the user install it manually
+if ! python3 --version > /dev/null 2>&1
+then echo "python3 is not available, please install it manually before attempting to install echidna" && exit 1
+fi
+
+# install pip3 if not already installed
+if ! pip3 --version > /dev/null 2>&1
+then
+  echo "pip3 is not available, installing it now.."
+  curl https://bootstrap.pypa.io/get-pip.py | python3
+fi
+
+# install slither if not already installed
+if ! slither --version > /dev/null 2>&1
+then
+  echo "slither is not available, installing it now.."
+  pip3 install slither-analyzer --user
+fi
+
+echo "Installing Echidna v$version.."
+
+BASE_DIR=${XDG_CONFIG_HOME:-$HOME}
+ECHIDNA_DIR=${ECHIDNA_DIR-"$BASE_DIR/.echidna"}
+ECHIDNA_BIN_DIR="$ECHIDNA_DIR/bin"
+ECHIDNA_ARCHIVE_DIR="$ECHIDNA_DIR/v$version"
+ECHIDNA_ARCHIVE_PATH="$ECHIDNA_ARCHIVE_DIR/echidna.tar.gz"
+ECHIDNA_DIGEST_PATH="$ECHIDNA_ARCHIVE_DIR/echidna.sha256"
+TRUE_BIN_PATH="$ECHIDNA_ARCHIVE_DIR/echidna/echidna"
+BIN_PATH="$ECHIDNA_BIN_DIR/echidna"
+
+if [[ "$(uname)" == "Darwin" ]]
+then arch="MacOS"
+elif [[ "$(uname)" == "Linux" ]]
+then arch="Ubuntu"
+else echo "Unsupported system: $(uname)" && exit 1
+fi
+
+TARBALL_URL="https://github.com/crytic/echidna/releases/download/v$version/echidna-$version-$arch.tar.gz"
+DIGEST_URL="https://github.com/crytic/echidna/releases/download/v$version/echidna-$version-$arch.tar.gz.sha256"
```

Comment:
Our digests are not signed, so someone tampering with the releases may also modify the digest. Is this meant only as an integrity check? Otherwise hardcoding the hashes next to the version in the script may provide a bit more of certainty (unless someone tampers with the script as well)

Reply:
Just integrity, hardcoding the hashes seems like a better path forward 👍